File: Queue/Javascript/queue.js
class Queue {
    constructor() {
        this.queue = []
    }
    push(data) {
        this.queue.push(data)
    }
    printQu() {
        return this.queue;
    }
    pop() {
        this.queue.shift()
    }
    front() {
        return this.queue[0]
    }
    back() {
        return this.queue[this.queue.length - 1]
    }
    size() {
        return this.queue.length
    }
    empty() {
        return this.queue.length === 0
    }
}

const q = new Queue()
q.push('a')
q.push('b')
q.push('c')

q.pop()
console.log(q.back())
console.log(q.size())
console.log(q.empty())

File: Queue/C++/queue.cpp
#include <iostream>
#include <queue>
using namespace std;

int main()
{
    queue<int> q;

    q.push(1);
    q.push(2);
    q.push(3);

    cout << "queue size " << q.size() << endl;
    cout << "queue back " << q.back() << endl;

    while (!q.empty())
    {
        cout << q.front() << " ";
        q.pop();
    };
    return 1;
}